common/lib/xmodule/xmodule/assets/vertical/public/js/vertical_student_view.js

```diff
@@ -13,4 +13,23 @@ window.VerticalStudentView = function(runtime, element) {
             apiUrl: $bookmarkButtonElement.data('bookmarksApiUrl')
         });
     });
+    $(element).find('.vert').each(
+        function(idx, block) {
+            var blockKey = block.dataset.id;
+
+            if (block.dataset.completableByViewing === undefined) {
+                return;
+            }
+            if (blockKey) {
+                $.ajax({
+                    type: 'POST',
+                    url: runtime.handlerUrl(element, 'publish_completion'),
+                    data: JSON.stringify({
+                        block_key: blockKey,
+                        completion: 1.0
+                    })
+                });
```

Comment:
If I view a sequential like "Lesson 1 - Getting Started" in the demo course and click from the first vertical to another vertical and then back to the first, I see that it still POSTs a `publish_completion` event to the server - it has "forgotten" that it already marked the block as complete.

I was going to suggest calling `delete block.dataset.completableByViewing;` after the AJAX request completes successfully, but I think that each vertical gets rendered from an HTML string when switching verticals in a sequential, so that approach wouldn't work.

If you think it's worth doing, maybe add a global variable to track already-submitted completion events? Or we can ignore the duplicates, as it's not a huge optimization.